src/main/java/net/openhft/chronicle/bytes/internal/SingletonEmptyByteStore.java

```diff
@@ -0,0 +1,505 @@
+        Ints.requireNonNegative(length);
+        throw newUnsupportedOperationException();
+    }
+
+    @NotNull
+    @Override
+    public EmptyByteStore write(@NonNegative final long writeOffset,
+                                @NotNull final RandomDataInput bytes,
+                                @NonNegative final long readOffset,
+                                @NonNegative final long length) {
+        requireNonNegative(writeOffset);
+        ReferenceCountedUtil.throwExceptionIfReleased(bytes);
+        requireNonNegative(readOffset);
+        requireNonNegative(length);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public byte readByte(@NonNegative long offset) {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public int peekUnsignedByte(@NonNegative long offset) {
+        requireNonNegative(offset);
+        return -1;
+    }
+
+    @Override
+    public short readShort(@NonNegative long offset) {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public int readInt(@NonNegative long offset) {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public long readLong(@NonNegative long offset) {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public float readFloat(@NonNegative long offset) {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public double readDouble(@NonNegative long offset) {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public byte readVolatileByte(@NonNegative long offset)
+            throws BufferUnderflowException {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public short readVolatileShort(@NonNegative long offset)
+            throws BufferUnderflowException {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public int readVolatileInt(@NonNegative long offset)
+            throws BufferUnderflowException {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public long readVolatileLong(@NonNegative long offset)
+            throws BufferUnderflowException {
+        requireNonNegative(offset);
+        throw newUnsupportedOperationException();
+    }
+
+    @Override
+    public boolean isDirectMemory() {
+        return false;
+    }
+
+    @NotNull
+    @Override
+    public EmptyByteStore copy() {
+        return this;
+    }
+
+    @Override
+    public @NotNull Bytes<Void> bytesForRead() throws IllegalStateException {
+        return new VanillaBytes<Void>(this, 0, 0) {
```

Comment:
I think this constructor should only be used by descendants of VanillaBytes, can we not push the envelope here?

Reply:
Conveniently, there is a `VanillaBytes.vanillaBytes()` that does this. I will fix it...